python/tank/bootstrap/configuration.py

```diff
@@ -78,18 +78,21 @@ def get_tk_instance(self, sg_user):
         path = self._path.current_os
         core_path = get_core_python_path_for_config(path)
 
+        # Get the user before the core swapping.
+        from ..authentication import serialize_user
+
         # swap the core out
         CoreImportHandler.swap_core(core_path)
 
+        log.debug("Core swapped, authenticated user will be set.")
+
+        self._set_authenticated_user(sg_user, serialize_user(sg_user))
```

Comment:
could the serialize_user happen before the core swap? Maybe i am over thinking it, but i feel that's a little tiny bit clearer. e.g.

```python
# Get the user before the core swapping.
from ..authentication import serialize_user
serialialized_user = serialize_user(sg_user)

CoreImportHandler.swap_core(core_path)

self._set_authenticated_user(serialialized_user)


```

Reply:
i really think we shouldn't pass *ANY* objects across the swap boundary here if possible. Can we not pass `sg_user` but purely rely on the serialized? that would be great.